src/gnome/dialog-imap-editor.c

```diff
@@ -0,0 +1,759 @@
+    if (type != imap_dialog->type)
+    {
+        imap_dialog->type = type;
+        get_account_info (imap_dialog);
+    }
+}
+
+static void
+show_count_column (ImapDialog *imap_dialog, gboolean show)
+{
+    GtkTreeViewColumn *tree_column;
+
+    // Show Count Column
+    tree_column = gtk_tree_view_get_column (GTK_TREE_VIEW(imap_dialog->view), 4);
+    gtk_tree_view_column_set_visible (tree_column, show);
+
+    // Hide Based on Column
+    tree_column = gtk_tree_view_get_column (GTK_TREE_VIEW(imap_dialog->view), 1);
+    gtk_tree_view_column_set_visible (tree_column, !show);
+
+    gtk_tree_view_columns_autosize (GTK_TREE_VIEW(imap_dialog->view));
+}
+
+static void
+add_to_store (GtkTreeModel *model, GtkTreeIter *iter, const gchar *text, gpointer user_data)
+{
+    gchar       *fullname = NULL;
+    gchar       *map_fullname = NULL;
+
+    struct imap_info *imapInfo = (struct imap_info*)user_data;
+
+    fullname = gnc_account_get_full_name (imapInfo->source_account);
+
+    // Do we have a valid map account
+    if (imapInfo->map_account == NULL)
+        map_fullname = g_strdup (_("Map Account NOT found"));
+    else
+        map_fullname = gnc_account_get_full_name (imapInfo->map_account);
+
+    PINFO("Add to Store: Source Acc '%s', Match '%s', Map Acc '%s'", fullname, imapInfo->match_string, map_fullname);
+
+    gtk_tree_store_set (GTK_TREE_STORE(model), iter,
+                        SOURCE_FULL_ACC, fullname, SOURCE_ACCOUNT, imapInfo->source_account,
+                        BASED_ON, text,
+                        MATCH_STRING, imapInfo->match_string,
+                        MAP_FULL_ACC, map_fullname, MAP_ACCOUNT, imapInfo->map_account,
+                        FULL_CATEGORY, imapInfo->full_category, COUNT, imapInfo->count,
+                        FILTER, TRUE, -1);
+
+    g_free (fullname);
+    g_free (map_fullname);
+}
+
+static void
+get_imap_info (Account *acc, const gchar *category, GtkTreeModel *model, const gchar *text)
+{
+    GtkTreeIter  toplevel, child;
+    GList *imap_list, *node;
+    gchar *acc_name = NULL;
+
+    acc_name = gnc_account_get_full_name (acc);
+    PINFO("Source Acc '%s', Based on '%s', Path Head '%s'", acc_name, text, category);
+
+    if (category == NULL) // For Bayesian, category is NULL
+        imap_list = gnc_account_imap_get_info_bayes (acc);
+    else
+        imap_list = gnc_account_imap_get_info (acc, category);
+
+    if (g_list_length (imap_list) > 0)
+    {
+        PINFO("List length is %d", g_list_length (imap_list));
+
+        // Add top level entry of Source full Account and Based on.
+        gtk_tree_store_append (GTK_TREE_STORE(model), &toplevel, NULL);
+        gtk_tree_store_set (GTK_TREE_STORE(model), &toplevel,
+                        SOURCE_ACCOUNT, acc, SOURCE_FULL_ACC, acc_name, BASED_ON, text, FILTER, TRUE, -1);
+
+        for (node = imap_list;  node; node = g_list_next (node))
+        {
+            struct imap_info *imapInfo;
+
```

Comment:
Declare and initialize in one statement if you can. It reduces the risk of having a path where you didn't initialize.